fuse.py

```diff
@@ -376,18 +377,27 @@ class fuse_operations(Structure):
         ('flag_reserved', c_uint, 29),
     ]
 
+# When importing "fuse", you can check this with "hasattr(fuse, 'NANOSECOND_INT_AVAILABLE')"
+NANOSECOND_INT_AVAILABLE = True
 
-def time_of_timespec(ts):
-    return ts.tv_sec + ts.tv_nsec / 10 ** 9
+def time_of_timespec(ts, nanosecond_int = False): # nanosecond_int is opt-in by default
+    if nanosecond_int:
+        return int(ts.tv_sec) * 10 ** 9 + int(ts.tv_nsec) # NEW STYLE: returns time as int in nanoseconds
+    else:
+        return ts.tv_sec + ts.tv_nsec / 10 ** 9 # OLD STYLE: returns time as float in seconds
 
-def set_st_attrs(st, attrs):
+def set_st_attrs(st, attrs, nanosecond_int = False): # nanosecond_int is opt-in by default
     for key, val in attrs.items():
         if key in ('st_atime', 'st_mtime', 'st_ctime', 'st_birthtime'):
             timespec = getattr(st, key + 'spec', None)
             if timespec is None:
                 continue
-            timespec.tv_sec = int(val)
-            timespec.tv_nsec = int((val - timespec.tv_sec) * 10 ** 9)
+            if nanosecond_int: # NEW STYLE: expects time as an int in nanoseconds
+                timespec.tv_sec = int(floor(val / 10 ** 9))
```

Comment:
This module imports [`from __future__ import division`](https://docs.python.org/3/library/__future__.html). Because of that, you should not use `floor`. You should instead use the double slash operator `//`. This has a similar effect, except that instead of casting to a `float` (like `floor` does), it preserves generates an `int` result from two `int` arguments. This should avoid any possible rounding errors that might come up from using `floor`. It also removes the need to import anything from the `math` module.